Purpur init

heisfer opened the pull request:
resolves  #37

## Things Done
- [x] Updated all purpur builds lock file
- [x] Updated Readme   
- [x]  `nix fmt`
- [x]  `nix flake check` 
- [x] Tested `pkgs.purpurServers.purpur` in own config
## Test code
```nix
{ inputs, pkgs, ... }:
{
  imports = [ inputs.nix-minecraft.nixosModules.minecraft-servers ];
  nixpkgs.overlays = [ inputs.nix-minecraft.overlay ];

  services.minecraft-servers = {
    enable = true;
    eula = true;
    servers = {
      purpur = {
        enable = true;
        package = pkgs.purpurServers.purpur;
      };
    };
  };
}

```

## Notes
- Purpur api doesn't store sha256
- Package is modified version of paper package
- `nix fmt` formatted other unrelated files

Infinidoge commented:
Please revert all formatting changes, they cause a lot of noise in an otherwise clean and straight forward PR.

Additionally, is there any reason why the update script is written to be async? I can't really see any reason why it should be, and it adds unnecessary complexity.

heisfer commented:
> Please revert all formatting changes, they cause a lot of noise in an otherwise clean and straight forward PR.

Sure thing!

> Additionally, is there any reason why the update script is written to be async? I can't really see any reason why it should be, and it adds unnecessary complexity.

The reason for that is fastly download old build and generate sha256 hash (since they only support md5 and I believe md5 is deprecated in nix). I'll revert it back to non async since it's no longer needed.

heisfer commented:
Things changed:
1. Reverted `update.py`
2. Added bad.json so we don't request not found builds each time.
3. Rechecked java versions.
4. Added `log4j.xml` for specific versions. 
5. Formated with `nixfmt-rfc-style` as it says in `flake.nix` (CONTRIBUTING.MD says nixpkgs-fmt)

I hope commit message is fit for [CONTRIBUTING.md](https://github.com/Infinidoge/nix-minecraft/blob/master/CONTRIBUTING.md)

Infinidoge commented on pkgs/purpur-servers/default.nix:
```diff
@@ -0,0 +1,71 @@
+    # Version older than 1.1 1.1 = false
+    if versionOlder v "1.17" then
+      jdk11
+    else if versionOlder v "1.18.2" then # paper says 1.18.1+ but 1.18.1 max is 17
```
The behavior doesn't seem to match the comment

heisfer replied:
jar files seems to have hard limit on max version
`Unsupported Java detected (61.0). Only up to Java 16 is supported.` - 1.16.5
`Unsupported Java detected (65.0). Only up to Java 17 is supported.` - 1.18.1 
So I'll leave it to that.

heisfer replied:
Another note is that 1.17 version seems to be completly broken. 
Doesn't support jdk11 and 17 is unsupported.

Infinidoge reviewed (commented):
Overall looks good to me, small question with regards to the java version selector

fmbearmf commented:
any news on the status of this?

Infinidoge commented:
Apologies for taking so long to get back to this. Would it be possible to skip including the `url` in the lockfile, since the URL is predictable based on the Minecraft version and the build number? That would cut down on a lot of lines in the lockfile.
